pandas/core/indexes/datetimes.py

```diff
@@ -316,12 +316,12 @@ def _simple_new(cls, values, name=None, freq=None, tz=None, dtype=None):
                 tz = validate_tz_from_dtype(dtype, tz)
                 dtype = DatetimeTZDtype(tz=tz)
             elif dtype is None:
-                dtype = _NS_DTYPE
+                dtype = values.dtype
 
             values = DatetimeArray(values, freq=freq, dtype=dtype)
             tz = values.tz
             freq = values.freq
-            values = values._data
+            values = values._data.view('i8')
```

Comment:
Why change this to integer data?

(and I see you do a similar change a few times below)

Reply:
Yep, removing.